tests/unittests/test_datasource/test_azure.py

```diff
@@ -775,18 +876,16 @@ def test_crawl_metadata_on_reprovision_reports_ready(
         dsrc = self._get_ds(data)
         poll_imds_func.return_value = ovfenv
         dsrc.crawl_metadata()
-        self.assertEqual(1, report_ready_func.call_count)
+        self.assertEqual(1, m_report_ready.call_count)
 
     @mock.patch('cloudinit.sources.DataSourceAzure.util.write_file')
     @mock.patch('cloudinit.sources.helpers.netlink.'
                 'wait_for_media_disconnect_connect')
     @mock.patch(
         'cloudinit.sources.DataSourceAzure.DataSourceAzure._report_ready')
-    @mock.patch('cloudinit.net.dhcp.EphemeralIPv4Network')
-    @mock.patch('cloudinit.net.dhcp.maybe_perform_dhcp_discovery')
```

Comment:
These mocks are no longer necessary
 Both `EphemeralIPv4Network` and `maybe_perform_dhcp_discovery` were previously being invoked and used within `EphemeralDHCPv4WithReporting` and `EphemeralDHCPv4`.
`EphemeralDHCPv4WithReporting` and `EphemeralDHCPv4` are now being mocked for all of `TestAzureDataSource`.